src/plugins/lua/replies.lua

```diff
@@ -91,16 +91,127 @@ local function replies_check(task)
           in_reply_to, stored_rcpt)
     end
 
-    return false
+    return nil
+  end
+
+  local function add_to_global_replies_set(params, sender_key)
+    lua_util.debugm(N, task, 'Adding recipients %s to global replies set', params)
+
+    local function zadd_global_set_cb(err, data)
+      if err ~= nil then
+        rspamd_logger.errx(task, 'failed to add sender %s to global replies set with error: %s', sender_key, err)
+        return
+      end
+      rspamd_logger.infox(task, 'added sender %s to global set with code: %s', sender_key, data)
+    end
+
+    table.insert(params, 1, 'rsrk_verified_recipients')
+
+    lua_redis.redis_make_request(task, -- making global replies set for verified recipients
+            redis_params, -- connect params
+            sender_key, -- hash key
+            true, -- is write
+            zadd_global_set_cb, --callback
+            'ZADD', -- command
+            params -- arguments
+    )
```

Comment:
Hum, I also don't see any limits for this key, it must have not more than X (configurable parameter - default should be like 20-30 at max) and `zrembyrank` command must be emitted after any successful write.

Reply:
Wdym I need to execute `zrembyrank` command after successful write? I just added recipients and I need to remove them or what? I don't really understand.

Reply:
You need to maintain the number of elements in the lists. Otherwise, they will grow infinitely and blow up the Redis instance. I would suggest to use the redis scripts here: they can do zadd/zcard/zrembyrank/expire all at once in the atomic execution pipeline.